src/ray/core_worker/core_worker.cc

```diff
@@ -1991,13 +1997,9 @@ std::vector<rpc::ObjectReference> CoreWorker::SubmitActorTask(
   } else {
     returned_refs = task_manager_->AddPendingTask(
         rpc_address_, task_spec, CurrentCallSite(), actor_handle->MaxTaskRetries());
-    io_service_.post(
-        [this, task_spec]() {
-          RAY_UNUSED(direct_actor_submitter_->SubmitTask(task_spec));
-        },
-        "CoreWorker.SubmitActorTask");
+    direct_actor_submitter_->SubmitTask(task_spec);
   }
-  return returned_refs;
+  return {returned_refs};
```

Comment:
Doesn't this occur copy?

Reply:
from returned_refs -> optional data structure. Or is it optimized by the compiler?

Reply:
Add std::move to avoid copy.